### PR TITLE
System.text.json support

### PR DESCRIPTION
**Pull request by andrueastman**
This PR is part of https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/issues/80

### CODE REVIEW

File: src/Microsoft.Graph/Generated/callrecords/requests/CallRecordSessionsCollectionResponse.cs [hunk 1]
```diff
@@ -9,25 +9,23 @@
...
     /// The type CallRecordSessionsCollectionResponse.
     /// </summary>
-    [JsonObject(MemberSerialization = MemberSerialization.OptIn)]
     public class CallRecordSessionsCollectionResponse
     {
         /// <summary>
         /// Gets or sets the <see cref="ICallRecordSessionsCollectionPage"/> value.
         /// </summary>
-		[JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName ="value", Required = Newtonsoft.Json.Required.Default)]
+		[JsonPropertyName("value")]
```
Comment (MIchaelMainer) [inline src/Microsoft.Graph/Generated/callrecords/requests/CallRecordSessionsCollectionResponse.cs:22]: nit: indent is off

**Review by MIchaelMainer — APPROVED** [review]
Sample reviewed these changes. Used `git diff dev -I "\s+using\s.*" -I "\s+\[JsonObject.*" -I "\s+\[JsonProperty.*" -I "\s+\[JsonConverter.*" -I "\s+\[JsonExtensionData.*" -I "\s+\[InterfaceConverter.*" -I "\s+\[JsonPropertyName.*" -I "\s+public\sSystem.Text.Json.JsonDocument.*" --ignore-space-change` to cut back on some of the repetitive review.

For some reason it didn't ignore this.

-        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "frequency", Required = Newtonsoft.Json.Required.Default)]
-        public Newtonsoft.Json.Linq.JToken Frequency { get; set; }
+        [JsonPropertyName("frequency")]
+        public System.Text.Json.JsonDocument Frequency { get; set; }

I also reviewed the output from the change in EntityReferenceRequest.cs.tt for writing @odata.id.

One note, GitHub only loads the first 3000 files out the 4463 files it says are changed. The local diff tool I'm using, sdvdiff, only shows 2857 files changed. I performed the diff against dev and local dev is up tp date. I don't know why there is that discrepancy. One or both of tools are wrong, or the operator is in error.

**Comment by MIchaelMainer** [comment]
Will you create a feature2.0 branch for this PR to pull into?

**Comment by andrueastman** [comment]
Thanks for pointing that out. I have changed the target branch to feature/2.0 branch which already exists.  
I will also look into fixing the indentation issues you have pointed out.

**Comment by andrueastman** [comment]
https://github.com/microsoftgraph/msgraph-sdk-dotnet/pull/870/commits/01fa5093689721cf5d6cc91b92bd04c443d5ae4b Resolves indentation issues.

File: src/Microsoft.Graph/Generated/callrecords/requests/CallRecordSessionsCollectionResponse.cs (hunk 1 above)
Comment (andrueastman) [reply src/Microsoft.Graph/Generated/callrecords/requests/CallRecordSessionsCollectionResponse.cs]: https://github.com/microsoftgraph/msgraph-sdk-dotnet/pull/870/commits/01fa5093689721cf5d6cc91b92bd04c443d5ae4b Resolves indentation issues.

**Pull request merged**